src/nativeapm/elements/inputs/text.ts

```diff
@@ -0,0 +1,202 @@
+     * @type {HTMLInputElement}
+     */
+    inputElement: HTMLInputElement;
+
+    /**
+     * Native APM input data
+     * @type {NativeApmInputData}
+     */
+    inputData: NativeApmInputData;
+
+    /**
+     * Native APM error message element
+     * @type {HTMLElement}
+     */
+    errorMessageElement: HTMLElement;
+
+    /**
+     * Native APM input label element
+     * @type {HTMLElement}
+     */
+    inputLabel: HTMLElement;
+
+    /**
+     * Theme of the Native APM form
+     * @type {NativeApmThemeConfigType}
+     */
+    theme: NativeApmThemeConfigType;
+
+    /**
+     * Native APM Input constructor
+     */
+    constructor(
+      inputData: NativeApmInputData,
+      theme: NativeApmThemeConfigType,
+      prefilledValue?: string
+    ) {
+      this.theme = theme;
+      this.inputData = inputData;
+      this.inputElement = this.createInputElement(prefilledValue);
+      this.inputLabel = this.createInputLabel();
+      this.errorMessageElement = this.createErrorMessageElement();
+    }
+
+    /**
+     * This function returns the input element
+     */
+    public getInputElement() {
+      const inputWrapper = document.createElement('div');
+
+      inputWrapper.setAttribute('class', 'native-apm-text-input-wrapper');
+
+      StylesUtils.styleElement(inputWrapper, this.theme.form.inputs.wrapper);
+
+      inputWrapper.appendChild(this.inputLabel);
+      inputWrapper.appendChild(this.inputElement);
+      inputWrapper.appendChild(this.errorMessageElement);
+
+      return inputWrapper;
+    }
+
+    /**
+     * This function validates the input
+     */
+    public isValid() {
+      const isRequiredPassed = this.inputData.required
+        ? this.inputElement.value.length > 0
+        : true;
+
+      const maxLengthPassed = this.inputData.max_length
+        ? this.inputElement.value.length <= this.inputData.max_length
+        : true;
+
+      const lengthPassed = this.inputData.length
+        ? this.inputElement.value.length === this.inputData.length
+        : true;
+
+      if (!(isRequiredPassed && maxLengthPassed && lengthPassed)) {
+        this.setErrorMessage();
+      }
+
+      return isRequiredPassed && maxLengthPassed && lengthPassed;
+    }
+
+    /**
+     * This function returns value of the input
+     */
+    public getValue() {
+      return { [this.inputData.key]: this.inputElement.value };
+    }
+
+    /**
+     * This function returns the input element
+     */
+    protected getRawInputElement() {
+      return this.inputElement;
+    }
+
+    /**
+     * This function creates input element
+     */
+    private createInputElement(prefilledValue?: string) {
+      const input = document.createElement('input');
+
+      input.setAttribute('class', 'native-apm-text-input');
+      input.setAttribute('name', this.inputData.key);
+
+      if (prefilledValue) {
+        input.value = prefilledValue;
+      }
+
+      input.addEventListener('keypress', this.resetErrorMessage.bind(this));
```

Comment:
Not a change request, just general comment.

I noticed there are several places uses `addEventListener` but without explicitly clean up the listener. In general it should be OK if the DOM element is short-lived. Once the `input` element is removed from DOM tree the memory will be free by GC.

I guess it might be difficult to remove listener in your case, but if you could find a way to remove the listener explicitly that would be great.

Reply:
I know that it's not ideal and we should always clean this up, but like you said in that case it will be difficult since we can explicitly manage the lifecycle of the class instance like in React for example and taking in consideration that those elements are indeed short-lived I would leave it as it is for now